### ISSUE TITLE
Error with binary_sensor.ble_motion in 7.8.2

### ISSUE THREAD

**Issue by Kaibob2** [custom-components/ble_monitor#739]
With Version 7.8.2 I get the following error for both MJYD02YL in my home.
Is there something I'm missing?

```
2022-02-22 17:36:21 ERROR (MainThread) [homeassistant.helpers.entity] Update for binary_sensor.ble_motion_bewegungsmelder_flur fails
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 521, in async_update_ha_state
await self.async_device_update()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 725, in async_device_update
raise exc
File "/config/custom_components/ble_monitor/binary_sensor.py", line 470, in async_update
if now - self._start_timer >= timedelta(seconds=self._reset_timer):
TypeError: unsupported operand type(s) for -: 'datetime.datetime' and 'str'
```

**Comment by Ernst79** [comment]
Did you get this message multiple times (for one sensor), or only once?

The 7.8.2 update could cause this error, but I think it will only happen once. So, if it doesn't happen again, you can just ignore it.

**Comment by Kaibob2** [comment]
I first thought, it happens each time the sensor detects any motion, but it appears, that it only happens 4 times
It's three hours later now with dozens correctly detected motions events but no more log entry.
I guess this will be `ignore` then :)

```
2022-02-22 17:33:06 ERROR (MainThread) [homeassistant.helpers.entity] Update for binary_sensor.ble_motion_bewegungsmelder_bad fails
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 521, in async_update_ha_state
await self.async_device_update()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 725, in async_device_update
raise exc
File "/config/custom_components/ble_monitor/binary_sensor.py", line 470, in async_update
if now - self._start_timer >= timedelta(seconds=self._reset_timer):
TypeError: unsupported operand type(s) for -: 'datetime.datetime' and 'str'
2022-02-22 17:33:21 ERROR (MainThread) [homeassistant.helpers.entity] Update for binary_sensor.ble_motion_bewegungsmelder_flur fails
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 521, in async_update_ha_state
await self.async_device_update()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 725, in async_device_update
raise exc
File "/config/custom_components/ble_monitor/binary_sensor.py", line 470, in async_update
if now - self._start_timer >= timedelta(seconds=self._reset_timer):
TypeError: unsupported operand type(s) for -: 'datetime.datetime' and 'str'
2022-02-22 17:36:21 ERROR (MainThread) [homeassistant.helpers.entity] Update for binary_sensor.ble_motion_bewegungsmelder_flur fails
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 521, in async_update_ha_state
await self.async_device_update()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 725, in async_device_update
raise exc
File "/config/custom_components/ble_monitor/binary_sensor.py", line 470, in async_update
if now - self._start_timer >= timedelta(seconds=self._reset_timer):
TypeError: unsupported operand type(s) for -: 'datetime.datetime' and 'str'
2022-02-22 17:41:21 ERROR (MainThread) [homeassistant.helpers.entity] Update for binary_sensor.ble_motion_bewegungsmelder_flur fails
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 521, in async_update_ha_state
await self.async_device_update()
File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 725, in async_device_update
raise exc
File "/config/custom_components/ble_monitor/binary_sensor.py", line 470, in async_update
if now - self._start_timer >= timedelta(seconds=self._reset_timer):
TypeError: unsupported operand type(s) for -: 'datetime.datetime' and 'str'
```

**Comment by Ernst79** [comment]
For now, I close the issue, but if it happens again, let me know, by reopening this issue.

**Comment by ACrazyConcept** [comment]
I get the same error also with MJYD02YL on version 7.9.6.
I think it happens shortly after restarting HA (in docker). Although I haven't monitored it that closely.

```
2022-03-12 11:40:29 ERROR (MainThread) [homeassistant.helpers.entity] Update for binary_sensor.ble_motion_28d12737b661 fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 535, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 744, in async_device_update
    raise exc
  File "/config/custom_components/ble_monitor/binary_sensor.py", line 470, in async_update
    if now - self._start_timer >= timedelta(seconds=self._reset_timer):
TypeError: unsupported operand type(s) for -: 'datetime.datetime' and 'str'
```

**Comment by Ernst79** [comment]
Could you do me a favor. At line 475-476 in `binary_sensor.py`, you can find the following code

```python
            except KeyError:
                self._state = self._newstate
```

Change this to 

```python
            except KeyError:
                self._state = self._newstate
            except TypeError:
                self._state = self._newstate
                _LOGGER.error("motion reset timer fails, start timer is %s", self._extra_state_attributes["last_motion"])
```

It should add an error in the log when it happens, but will continue to work. Please post the error here, so I can fix it.

**Comment by ACrazyConcept** [comment]
`2022-03-12 15:40:21 ERROR (MainThread) [custom_components.ble_monitor.binary_sensor] motion reset timer fails, start timer is 2022-03-12T15:38:19.021859+01:00`

**Comment by janikarna** [comment]
I seem to have this problem also:

Logger: homeassistant.helpers.entity
Source: custom_components/ble_monitor/binary_sensor.py:470
Integration: Passive BLE monitor ([documentation](https://github.com/custom-components/ble_monitor), [issues](https://github.com/custom-components/ble_monitor/issues))
First occurred: 16.45.51 (379 occurrences)
Last logged: 17.18.09

Update for binary_sensor.ruuvitag_ulko_motion_cb2e472bead8 fails
Update for binary_sensor.ruuvitag_tyohuone_motion_fa2e97d9f02e fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 535, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 744, in async_device_update
    raise exc
  File "/config/custom_components/ble_monitor/binary_sensor.py", line 470, in async_update
    if now - self._start_timer >= timedelta(seconds=self._reset_timer):
TypeError: unsupported operand type(s) for -: 'datetime.datetime' and 'str'

I have Home Assistant OS running on RPI4. I don't think I'll be able to edit any .py files.

**Comment by Ernst79** [comment]
I'll fix it in the next release. Somehow, the self._start_timer is stored as a string, not a datetime.datetime object.

**Comment by janikarna** [comment]
Ok, thank you for the information. the sensor does work as expected and the errors are not piling up for a while if I go and move the sensors a bit. But mine is for temp/humidity measurements and stationary, so motion is not needed :)

**Comment by Ernst79** [comment]
Should be fixed in 8.1.0, which has just been released.
